src/Hpack/Config.hs

```diff
@@ -468,17 +468,16 @@ hasKey key (Object o) = HashMap.member key o
 hasKey _ _ = False
 
 newtype Condition = Condition {
-  _conditionCondition :: Cond
+  conditionCondition :: Cond
 } deriving (Eq, Show, Generic, FromValue)
 
-newtype Cond = Cond String
+data Cond = CondBool Bool | CondExpression String
   deriving (Eq, Show)
 
 instance FromValue Cond where
   fromValue v = case v of
-    String s -> return (Cond $ T.unpack s)
-    Bool True -> return (Cond "true")
-    Bool False -> return (Cond "false")
+    String c -> return (CondExpression $ T.unpack c)
```

Comment:
I think this means someone could avoid this new filtering behavior by explicitly passing a string. Like this:

``` yaml
when:
  - condition: 'false'
```

That's probably fine, and may in fact be desired as a workaround.

Reply:
Yes, exactly, that's what I thought.